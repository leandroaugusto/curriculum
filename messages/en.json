{
  "Index": {
    "header": {
      "role": "Senior Frontend Developer | React JS | React Native",
      "actions": "Print - Share"
    },
    "about": {
      "title": "About",
      "description": "Front-end developer with 12 years of experience in digital business.",
      "locale": "Sao Paulo - Brazil",
      "list": {
        "phone": {
          "text": "+55 11 96566 4487",
          "link": "https://wa.me/5511965664487",
          "icon": "PhoneIcon"
        },
        "email": {
          "text": "leandroacosta2008@gmail.com",
          "link": "mailto:leandroacosta2008@gmail.com",
          "icon": "LetterIcon"
        },
        "linkedin": {
          "text": "linkedin.com/in/leandroaugustocosta",
          "link": "https://www.linkedin.com/in/leandroaugustocosta/?locale=en_US",
          "icon": "DocumentIcon"
        },
        "github": {
          "text": "github.com/leandroaugusto",
          "link": "https://github.com/leandroaugusto",
          "icon": "CodeIcon"
        }
      }
    },
    "academic": {
      "title": "Academic Education",
      "description": "Digital and Multimedia Design Anhembi Morumbi University"
    },
    "skills": {
      "title": "Skills",
      "list": {
        "react": {
          "name": "React JS"
        },
        "typescript": {
          "name": "Typescript"
        },
        "preProcessors": {
          "name": "Pre-processors"
        },
        "leadership": {
          "name": "Leadership"
        },
        "cssInJs": {
          "name": "CSS-in-JS"
        },
        "designPatterns": {
          "name": "Design Patterns"
        },
        "agile": {
          "name": "Agile Methodologies"
        },
        "microfrontend": {
          "name": "Microfrontend"
        },
        "git": {
          "name": "Code versioning"
        },
        "tests": {
          "name": "Unit/E2E Tests"
        }
      }
    },
    "experience": {
      "title": "Professional Experience",
      "list": {
        "doris": {
          "year": "2023-",
          "jobType": "Remote",
          "name": "Doris Mobi",
          "role": "Senior Frontend Developer",
          "description": "I'm working at a company that created and maintains a virtual dressing room widget, using artificial intelligence. I work with frontend, participating in the team's day-to-day activities, developing different projects within the company. I use React JS, React Native, Styled-components, Typescript, Node JS, GraphQL, React Testing Library, Cypress, Circle CI, Sentry and everyday company tools."
        },
        "united": {
          "year": "2023-2023 (4 months)",
          "jobType": "Remote",
          "name": "United Futsal",
          "role": "Fullstack Developer",
          "description": "I worked on an international project, for a client based in the USA, with a team made up of Brazilians. I was responsible for planning, discussing and developing the architecture, structure and content of the frontend and backend code for a platform that organizes futsal events, adapting it to a SaaS business model. I used React JS, Typescript, Styled-components, Redux, Kotlin, Quarkus, Docker and MongoDB"
        },
        "dasa": {
          "year": "2022-2023",
          "jobType": "Remote",
          "name": "Genera - DASA Group",
          "role": "Senior Frontend Developer",
          "description": "Integrated into the technical team, I worked with web development, using the Wordpress platform and the Gutenberg plugin, based on React. I used React JS, PHP and everyday company tools."
        },
        "soma": {
          "year": "2021-2022",
          "jobType": "Remote",
          "name": "SOMA Group",
          "role": "Mobile Developer | React Native",
          "description": "Working on the client application with maintenance and updates. Actively participating in discussions about internal improvements in the company's technical team. Directly responsible for maintaining and implementing new features in the OffPremium application. In the first months, I worked alone in this role and later I worked as a tech lead, leading 3 more people with Jr. level in maintaining the app. I used React Native, GraphQL, SWR, Redux, Typescript, Styled-components, Sentry, VTEX Platform, Appsflyer, Firebase and other tools related to Apps."
        },
        "smiles": {
          "year": "2018-2021",
          "jobType": "Hybrid",
          "name": "Smiles Argentina",
          "role": "Senior Frontend Developer",
          "description": "Responsible for developing and maintaining the Smiles portal for Argentina. I also participated in the development of landing pages and maintenance on the Brazil portal. I worked on a multidisciplinary technical team, and with intense contact with Smiles leaders. I used React JS, Typescript, Bootstrap, Sass, and Axios"
        },
        "mira": {
          "year": "2017-2018",
          "jobType": "On Site",
          "name": "Mira Education",
          "role": "Full Frontend Developer",
          "description": "Front-end responsible for developing the portal/web version in ReactJS of the Mira Aula application, integrated into a team with 2 other Front developers, as well as Back-end and mobile developers. Used React JS, Redux, GraphQL, AWS, Jenkins, Apollo, Nigthwatch, Node JS and Jest with Enzyme"
        },
        "fillet": {
          "year": "2016-2017",
          "jobType": "On Site",
          "name": "Fillet Agency",
          "role": "Full Frontend Developer",
          "description": "Front-end responsible for developing websites and hotsites for several clients - starting my contact with ReactJS - integrated into a team with 9 other developers. I used React JS, Redux, Flux, HTML and CSS"
        },
        "isobar": {
          "year": "2014-2016",
          "jobType": "On Site",
          "name": "Click/Isobar Agency",
          "role": "Jr/Full Frontend Developer",
          "description": "Integrated into a team responsible for preparing and maintaining web projects for Isobar's various clients. I was able to participate in the processes, from the initial development stage, involving structuring folders and files and defining assets, to the sending stage for deployment. Often responsible for media tagging of projects. Always aligning directives, both with direct coordination and with the creative and business team. I used Javascript, HTML, CSS, Node JS, Less, Sass, Angular JS and PHP"
        },
        "novi": {
          "year": "2014-2014 (4 months)",
          "jobType": "On Site",
          "name": "Novi Agency",
          "role": "Jr Frontend Developer",
          "description": "Front-end development of websites, from institutional, portals to small e-commerces. I also worked on the development and conversion of Flash to HTML, of educational games for tablets. I used HTML, CSS and Javascript"
        },
        "kindle": {
          "year": "2011-2014",
          "jobType": "On Site",
          "name": "Kindle/GR8 Agency",
          "role": "Intern/Jr Frontend Developer",
          "description": "I started as an intern at the Kindle SP Agency, later being hired as a junior Front-end. I worked on maintaining websites and portals, in addition to participating in the entire development of tablet applications, from interactive games to Visual Aids. Currently the agency is in the events and interactivity business, under the name Gr8!. I used Photoshop, Illustrator, Sublime Text, Javascript, HTML, CSS, PHP, MySQL and tools typical of an agency's daily life."
        },
        "infservice": {
          "year": "2010-2011",
          "jobType": "On Site",
          "name": "Infservice",
          "role": "Intern",
          "description": "I interned at the company Infservice, with streaming technology, for a period of 1 year. I developed layouts for presentations, workshops and online lectures, video editing, HTML for marketing emails and slide synchronization, all managed by a tool called ON24. I used HTML, CSS, Office Package and specific tools from the company's daily routine."
        }
      }
    },
    "knowledge": {
      "title": "Knowledge"
    },
    "footer": {
      "print": "Print to pdf",
      "share": "Share"
    },
    "share": {
      "title": "Leandro Costa's CV | Senior Frontend",
      "email": "I would like to share with you the profile of Leandro Augusto da Costa, Senior Frontend Developer with experience in React JS and React Native."
    }
  }
}