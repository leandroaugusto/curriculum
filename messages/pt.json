{
  "Index": {
    "header": {
      "role": "Desenvolvedor Frontend Sênior | React JS | React Native",
      "actions": "Imprimir - Compartilhar"
    },
    "about": {
      "title": "Sobre",
      "description": "Desenvolvedor front-end com 12 anos de experiência em negócios digitais. Já trabalhei em agências de publicidade, empresas do ramo de educação, pontos de fidelidade, saúde e e-commerces. Atualmente, minha stack é React JS, Node JS, Typescript e seus derivantes. Gosto de tecnologia e busco sempre desenvolver soluções que ajudem as pessoas a completarem suas tarefas e concluírem seus objetivos com mais facilidade. Sou entusiasta de automação residencial e toco piano por hobby.",
      "locale": "São Paulo - SP",
      "list": {
        "phone": {
          "text": "(11) 96566 4487",
          "link": "https://wa.me/5511965664487",
          "icon": "PhoneIcon"
        },
        "email": {
          "text": "leandroacosta2008@gmail.com",
          "link": "mailto:leandroacosta2008@gmail.com",
          "icon": "LetterIcon"
        },
        "linkedin": {
          "text": "linkedin.com/in/leandroaugustocosta",
          "link": "https://www.linkedin.com/in/leandroaugustocosta",
          "icon": "DocumentIcon"
        },
        "github": {
          "text": "github.com/leandroaugusto",
          "link": "https://github.com/leandroaugusto",
          "icon": "CodeIcon"
        }
      }
    },
    "academic": {
      "title": "Formação Acadêmica",
      "description": "Desing Digital e Multimídia Universidade Anhembi Morumbi"
    },
    "skills": {
      "title": "Habilidades",
      "list": {
        "react": {
          "name": "React JS"
        },
        "typescript": {
          "name": "Typescript"
        },
        "preProcessors": {
          "name": "Pre-processadores"
        },
        "leadership": {
          "name": "Liderança"
        },
        "cssInJs": {
          "name": "CSS no JS"
        },
        "designPatterns": {
          "name": "Padrões de Projeto"
        },
        "agile": {
          "name": "Metodologias Ágeis"
        },
        "microfrontend": {
          "name": "Microfrontend"
        },
        "git": {
          "name": "Versionamento de código"
        },
        "tests": {
          "name": "Testes Unitários/E2E"
        }
      }
    },
    "experience": {
      "title": "Experiência Profissional",
      "list": {
        "doris": {
          "year": "2023-",
          "jobType": "Remoto",
          "name": "Doris Mobi",
          "role": "Desenvolvedor Frontend Sênior",
          "description": "Estou atuando em uma empresa que criou e mantém um widget de provador de roupas virtual, utilizando inteligência artificial. Atuo com frontend, participando do dia-a-dia do time, desenvolvendo em diferentes projetos da empresa. Utilizo React JS, React Native, Styled-components, Typescript, Node JS, GraphQL, React Testing Library, Cypress, Circle CI, Sentry e ferramentas do cotidiano da empresa."
        },
        "united": {
          "year": "2023-2023 (4 meses)",
          "jobType": "Remoto",
          "name": "United Futsal",
          "role": "Desenvolvedor Fullstack",
          "description": "Atuei em um projeto internacional, de cliente baseado nos EUA, com time formado por brasileiros. Fui responsável por planejar, discutir e desenvolver a arquitetura, estrutura e conteúdo do código frontend e backend para uma plataforma que organiza eventos de futsal, adaptando-a para um modelo de negócio SaaS. Utilizei React JS, Typescript, Styled-components, Redux, Kotlin, Quarkus, Docker e MongoDB"
        },
        "dasa": {
          "year": "2022-2023",
          "jobType": "Remoto",
          "name": "Genera - Grupo DASA",
          "role": "Desenvolvedor Frontend Sênior",
          "description": "Integrado no time técnico, atuei com desenvolvimento web, utilizando a plataforma Wordpress e o plugin Gutenberg, baseado em React. Utilizei React JS, PHP e ferramentas do cotidiano da empresa."
        },
        "soma": {
          "year": "2021-2022",
          "jobType": "Remoto",
          "name": "Grupo SOMA",
          "role": "Desenvolvedor Mobile | React Native",
          "description": "Trabalhando no aplicativo do cliente com manutenção e atualização. Participando ativamente de discussões acerca de melhorias internas no time técnico da empresa. Responsável direto pela manutenção e implementação de novas features no aplicativo OffPremium. Nos primeiros meses, atuei sozinho nessa função e posteriormente fui atuando como tech lead, liderando mais 3 pessoas com nível Jr. na manutenção do app. Utilizei React Native, GraphQL, SWR, Redux, Typescript, Styled-components, Sentry, Plataforma VTEX, Appsflyer, Firebase e outras ferramentas relativas a Apps."
        },
        "smiles": {
          "year": "2018-2021",
          "jobType": "Híbrido",
          "name": "Smiles Argentina",
          "role": "Desenvolvedor Frontend Sênior",
          "description": "Responsável pelo desenvolvimento e manutenção do portal Smiles para Argentina. Participei também do desenvolvimento de landing pages e manutenções no portal do Brasil. Atuei em um time técnico multidisciplinar, e com intenso contato com a lideranças da Smiles. Utilizei React JS, Typescript, Bootstrap, Sass, e Axios"
        },
        "mira": {
          "year": "2017-2018",
          "jobType": "Presencial",
          "name": "Mira Educação",
          "role": "Desenvolvedor Frontend Pleno",
          "description": "Front-end responsável pelo desenvolvimento da versão portal/web em ReactJS do aplicativo Mira Aula, integrado em uma equipe com outros 2 desenvolvedores Front, além de desenvolvedores Back-end e mobile. Utilizei React JS, Redux, GraphQL, AWS, Jenkins, Apollo, Nigthwatch, Node JS e Jest com Enzyme"
        },
        "fillet": {
          "year": "2016-2017",
          "jobType": "Presencial",
          "name": "Agência Fillet",
          "role": "Desenvolvedor Frontend Pleno",
          "description": "Front-end responsável pelo desenvolvimento de sites e hotsites de diversos clientes - iniciando meu contato com ReactJS - integrado em uma equipe com outros 9 desenvolvedores. Utilizei React JS, Redux, Flux, HTML e CSS"
        },
        "isobar": {
          "year": "2014-2016",
          "jobType": "Presencial",
          "name": "Agência Click/Isobar",
          "role": "Desenvolvedor Frontend Júnior/Pleno",
          "description": "Integrado em uma equipe responsável pela elaboração e manutenção de projetos web dos diversos clientes da Isobar. Pude participar dos processos, desde a etapa inicial de desenvolvimento, envolvendo estruturação de pastas e arquivos e definição de assets, até a etapa de envio para deploy. Responsável, muitas vezes, pelo tagueamento de mídia dos projetos. Sempre alinhando as diretivas, tanto com a coordenação direta, quanto com a equipe de criação e negócios. Utilizei Javascript, HTML, CSS, Node JS, Less, Sass, Angular JS e PHP"
        },
        "novi": {
          "year": "2014-2014 (4 meses)",
          "jobType": "Presencial",
          "name": "Agência Novi",
          "role": "Desenvolvedor Frontend Júnior",
          "description": "Desenvolvimento front-end de sites, desde institucionais, portais até pequenos e-commerces. Atuei também no desenvolvimento e conversão de Flash para HTML, de jogos educativos para tablet. Utilizei HTML, CSS e Javascript"
        },
        "kindle": {
          "year": "2011-2014",
          "jobType": "Presencial",
          "name": "Agência Kindle/GR8",
          "role": "Estagiário / Desenvolvedor Frontend Júnior",
          "description": "Comecei como estagiário na Agência Kindle SP, posteriormente sendo efetivado no cargo de Front-end junior. Atuei na manutenção de sites e portais, além de participar de todo o desenvolvimento de aplicações para tablet, desde jogos interativos até Visual Aids. Atualmente a agência está no ramo de eventos e interatividade, com o nome Gr8!. Utilizei Photoshop, Ilustrator, Sublime Text, Javascript, HTML, CSS, PHP, MySQL e ferramentas próprias do cotidiano de uma agência."
        },
        "infservice": {
          "year": "2010-2011",
          "jobType": "Presencial",
          "name": "Infservice",
          "role": "Estagiário",
          "description": "Estagiei na empresa Infservice, com a tecnologia de streaming, pelo período de 1 ano. Desenvolvi layouts para as apresentações, workshops e palestras online, edição de vídeos, HTML de emais marketing e sincronização de slides, tudo gerenciado por uma ferramenta de nome ON24. Utilizei HTML, CSS, Pacote Office e ferramentas específicas do cotidiano da empresa."
        }
      }
    },
    "knowledge": {
      "title": "Conhecimentos"
    },
    "footer": {
      "print": "Imprimir em pdf",
      "share": "Compartilhar"
    },
    "share": {
      "title": "Currículo de Leandro Costa | Frontend Sênior",
      "email": "Gostaria de compartilhar com você o perfil de Leandro Augusto da Costa, Desenvolvedor Frontend Sênior com experiência em React JS e React Native."
    }
  }
}