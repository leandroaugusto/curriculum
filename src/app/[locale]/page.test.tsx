import { render } from "@testing-library/react";
import { NextIntlClientProvider } from "next-intl";

import messages from "../../../messages/pt.json";
import Home from "./page";

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(),
  })
) as jest.Mock;

jest.mock("@/contexts/DialogContext", () => ({
  DialogProvider: () => (
    <div className="dialog-provider" data-testid="mock-dialog-provider" />
  ),
}));

describe("Home", () => {
  it("renders the Home page", () => {
    const { container } = render(
      <NextIntlClientProvider locale="en" messages={messages}>
        <Home />
      </NextIntlClientProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
