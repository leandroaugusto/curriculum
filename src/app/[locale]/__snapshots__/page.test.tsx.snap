// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the Home page 1`] = `
<div>
  <div
    class="dialog-provider"
    data-testid="mock-dialog-provider"
  />
</div>
`;
